### DISCUSSION TITLE
CasaOS is not an os

### DISCUSSION THREAD

**Discussion by daniel-widrick** [IceWhaleTech/CasaOS#218] [General]
So I've seen this project posted around several times now. Discussion universally devolves into confusion about the name.

Perhaps the project maintainers should review the name or lay out a roadmap to become a proper OS?

**Comment by aljaxus** [comment]
It's my first time even hearing about this project, but I'll give my two cents;

Since, as I understand it, this project is mean to be a simple solution for a small homelab *(not like multiple servers in a rack homelab, but a reused PC with a few additional drives in it serving as a simple media server, NAS, etc..)* why not base it off Alpine Linux?
You could basically just make an image that users can flash onto a USB and have a bootable USB drive.
Then you just create a simple web UI that adds few simple functions;
- updating the packages
- simple drive management
  - (un)mounting drives
  - format drive
  - mkfs.X drive

I think that using Alpine and just saving only the necessary files via LBU would make this a great, lightweight, fast and simple to setup tool.

**Comment by jerrykuku** [comment]
> It's my first time even hearing about this project, but I'll give my two cents;
> 
> Since, as I understand it, this project is mean to be a simple solution for a small homelab _(not like multiple servers in a rack homelab, but a reused PC with a few additional drives in it serving as a simple media server, NAS, etc..)_ why not base it off Alpine Linux? You could basically just make an image that users can flash onto a USB and have a bootable USB drive. Then you just create a simple web UI that adds few simple functions;
> 
> * updating the packages
> * simple drive management
>   
>   * (un)mounting drives
>   * format drive
>   * mkfs.X drive
> 
> I think that using Alpine and just saving only the necessary files via LBU would make this a great, lightweight, fast and simple to setup tool.

We will consider your suggestion, thank you.

**Comment by zhuwei** [comment]
that is right

**Comment by JohnGuan** [comment]
Very good question and you're right. CasaOS is not an "OS" yet, but we do aim to make it an "OS". As we said in README and Discord, it's still in the very early stages of CasaOS development.

We just don't want to keep our heads in the sand and wait until we've been working on it for a long time and say, "Hey, we've developed a home cloud system called CasaOS, do you want to try it?" . At this point, I don't think it's going to work much better than it does now.

We want to get it out to the community earlier, and talk to the community, discuss it, grow it, and then present it to more people and say, "Hey, our community built a CasaOS, and I can't believe you don't have one in your house. Everyone in our community is using it in their homes!"

You're also right about the roadmap, we don't have a clear roadmap yet, and that's a topic we'll be discussing here and in Discord, and it's becoming clearer after all this time. I think we'll draft one as soon as possible and discuss it together.

Thank you very much for your attention! You and everyone else are very welcome to suggest ideas that we can discuss together to make CasaOS the best choice for a Home Cloud system!😄

**Comment by tigerinus** [comment]
My understanding of OS is that it's a broader term and describes the nature of this project literally - It is a `system` to allow home lab (in team's term `family cloud`) to `operate` different server apps for home use. The currently implementation is an layer on top of the Ubuntu distro with dockerization. 

If we take any other OS we find it's technically the same idea - an layer of applications on top of a distro for specific scenarios. 

The app management could be done via native package management, call it `rpm` or `deb`, or dockerized package management which CasaOS embraces.

**Comment by JohnGuan** [comment]
This is an interesting and ongoing topic for some time, and I'm putting it into the discussion.

**Comment by Protektor-Desura** [comment]
If it were me I would recommend using the Ubuntu server or the Ubuntu cloud/core image as the basis for things. 

1... Ubuntu is the largest OS in the cloud by far. 

2... There is far more information and support out there for Ubuntu than any other Linux distro in my opinion. 

3... If you are worried about the size of the base/core OS that is why you could use the Ubuntu Cloud image as the starting point. There are many dockers that use the Ubuntu core image as the base for the docker to keep the size small.

It is also my opinion that Ubuntu is going to give you a fairly recent version of everything (libraries and drivers) without being on the bleeding edge where you can run in a lot of new problems because you are on the bleeding edge.

https://cloud-images.ubuntu.com/
https://ubuntu.com/download/server
https://hub.docker.com/r/multiarch/ubuntu-core/

**Reply by Hedius** [reply]
Ubuntu is not the most used server linux. Both centos/rhel and debian have a higher share on the market. Generally I would recommend to stay with Debian as a base.

You can have more up2date libraries in containers ofc. Furthermore, debian testing or unstable can be used as a base if needed.

**Reply by tigerinus** [reply]
For cloud use I'd recommend Alpine. My company builds 200+ microservices, all based on Alpine Linux, super light image footprint.

I use Debian Bullseye for development environment though. For one, it's the base of Ubuntu so for most non-Desktop issues can be resolved by simply following Ubuntu's troubleshooting guide. For second, it's lightweight (Alpine is even more lightweight but it's not for personal use) and stable (thanks for the long testing cycle).

In future if CasaOS would needs to turn into a Distro, I hope it's based off from Debian just like Ubuntu.

**Reply by Protektor-Desura** [reply]
> Ubuntu is not the most used server linux. Both centos/rhel and debian have a higher share on the market. Generally I would recommend to stay with Debian as a base.

Ubuntu accounts for 33.9% of the Linux market.
That means that over a third of websites that use Linux run on Ubuntu. Second place goes to Debian, which accounts for 16% of Linux-based websites, followed by CentOS with a share of 9.3%. Other distributions, like RedHat, Gentoo, Fedora, and SuSe, have a much smaller share, i.e., below 1%. Their shares are 0.8%, 0.5%, 0.2%, and 0.1%, respectively.
https://truelist.co/blog/linux-statistics/

Also Ubuntu works with Microsoft the most, especially on their Azure cloud.
Ubuntu is also the largest install base in the Amazon cloud as well
https://www.zdnet.com/article/ubuntu-linux-continues-to-rule-the-cloud/
https://recruit-c7ff.kxcdn.com/recruit/wp-content/uploads/2020/05/he-developer-survey-2020.pdf

The list just goes on with Ubuntu at the top in the cloud and most servers on the Internet running Linux.

**Reply by Hedius** [reply]
Not of the market of webservers where the OS is know. 

Only relying on webservers is biased.

Especially in the next statement of your source you can read "Red Hat has a 33.9% share in the worldwide server operating environment market." Dominating it together with windows. Straight ignored it.

Btw. Did you know that the cloud is not the whole server market? Only a part of it?

I know a ton of ubuntu free server environments.

But i will stop here now and not comment further. Useless discussion in the end.

**Reply by kerta1n** [reply]
I know this thread is about to become one year old, but I would honestly avoid Ubuntu. Snap is now included in all builds by default, which can make identifying drives and partitions much more difficult. I don't understand why Canonical had to make this change to Core instead of just Desktop, but it does not help in the case of server usage. I wouldn't go as far as saying Debian should be avoided either, however I do understand that some users may want the latest and greatest for drivers and use distros like Arch. Personally, I think it would be more efficient to have two branches of CasaOS: builds using DietPi, and builds using Alpine. Alpine is extremely lightweight compared to Debian, but doesn't have the familiarity that Debian does, so users can pick between the two.

Edit: I saw some screenshots of CasaOS being run on top of DietPi (which essentially is just a debloated build of debian [not fork!!]). I've edited the above reply to include DietPi.

**Comment by JohnGuan** [comment]
I'm not a Linux expert, but an OS enthusiast of sorts.
I've seen a lot of operating systems based on Debian, and I know of some operating systems that were originally based on Ubuntu that will be modified upstream to Debian.
Debian seems like a choice that you can't go wrong with.

Personally, I've basically switched from CentOS to Ubuntu for some of my simple cloud servers, which are of course personal hobbies and no production projects. I just felt a little more comfortable with Ubuntu, for reasons I can't quite put my finger on.
But by the time I was deploying ZimaBoard, the memory and storage usage made me more inclined to use Debian, but with a few modifications to Debian, such as adding sudo, changing the PATH, etc. To make Debian a little easier to use.
What struck me was that I had to add testing sources to fix it when I installed xrdp due to a delayed version update. (Ubuntu was updated a long time ago)

@Hedius I'm a bit interested in your opinion about Ubuntu, you seem to dislike it, can you tell me what Ubuntu has done to you to make you dislike it so much?

**Reply by kerta1n** [reply]
See https://github.com/IceWhaleTech/CasaOS/discussions/218#discussioncomment-5743288

**Comment by DocBrown101** [comment]
I would like to make one more suggestion. My server has been running Debian and Docker for years and all my services are defined with docker compose. It would be very helpful if there can be support for this from CasaOS.

Some kind of advanced operation with docker compose.

Then you're not just limited to the app store.

**Comment by taokanyunpi** [comment]
能否支持centos？

**Reply by Protektor-Desura** [reply]
CentOS technically does NOT exist any more. This was a huge controversy with Red Hat what they did with CentOS.

**Comment by c-shankar** [comment]
just like android roms are not really read only memories

**Comment by supermorph** [comment]
if it was me, i would use devuan, because i feel if we had this casaos in a form able to use without systemd, we could make it work on any Debian-like & non debian system, as it would not be required to use systemd but if it had it it could still work as it should.

i have a ubuntu image i custom made for backing up the server i made for home use, it does have casaos and is based on 25.04 (used to be 24.04 but i needed some things 24.x couldnt do at the time), but one day i went about migrating the os to Debian (as in doing an apt upgrade and install the keyrings and packages until the os became Debian and not ubuntu , im aware its non standard, but i know how to get around these issues so its a non issue for me as i was making it in a virtual machine) then i tried to make the same os build with devuan, and i like devuan, if i had that as a server os with casaos on, it would definitely be for me like the best of both worlds, as it wouldnt need bloat and i could still keep a desktop environment without all the un-necessary stuff ubuntu puts on.

and it didnt work as i thought, 
so i tried manually by making a devuan 5 vm, and installing, but definitely wouldn't install as no systemd. 
so i thought what about making calls to whatever sysv/systemd alternative's package that was installed 
as you see in package names when you install them and they are transitional packages that install something else, 

so why not use that theory to make it so we can use the environment that's installed? 
this way we can be installing on pretty much any Linux os

i know well more than just a simple idea, but it would make it distro agnostic and not locked to certain platforms 
(by such, i mean brands/types of distro that use systemd)
